src/core_init_atmosphere/mpas_init_atm_static.F

```diff
@@ -306,66 +323,127 @@ subroutine init_atm_static(mesh, dims, configs)
-          !
-          if (bdyMaskCell(iPoint) < nBdyLayers) then
-             ter(iPoint) = ter(iPoint) + rarray(i,j,1)
-             nhs(iPoint) = nhs(iPoint) + 1
-
-          ! For outermost boundary cells, additional work is needed to verify that the pixel
-          ! actually lies within the nearest cell
-          else
-             zPixel = sphere_radius * sin(lat_pt)                  ! Model cell coordinates assume a "full" sphere radius
-             xPixel = sphere_radius * cos(lon_pt) * cos(lat_pt)    ! at this point, so we need to ues the same radius
-             yPixel = sphere_radius * sin(lon_pt) * cos(lat_pt)    ! for source pixel coordinates
+ ierr = mgr % init(trim(config_geog_data_path)//trim(geog_sub_path))
+ if (ierr /= 0) then
+    call mpas_log_write('Error initializing ter Geotile Manager', messageType=MPAS_LOG_CRIT)
+    return
+ endif
 
-             if (in_cell(xPixel, yPixel, zPixel, xCell(iPoint), yCell(iPoint), zCell(iPoint), &
-                         nEdgesOnCell(iPoint), verticesOnCell(:,iPoint), xVertex, yVertex, zVertex)) then
 
-                ter(iPoint) = ter(iPoint) + rarray(i,j,1)
-                nhs(iPoint) = nhs(iPoint) + 1
+ allocate(ter_integer(nCells))
+ allocate(nhs(nCells))
 
-             end if
-          end if
-       end do
-       end do
+ !
+ ! Store tile values as a I8KIND integer temporarily to avoid floating
+ ! point round off differences
+ !
+ ter_integer(:) = 0_I8KIND
+ ter(:) = 0.0_RKIND
+ nhs(:) = 0
 
-    end do
+ call mpas_pool_get_config(mgr % pool, 'tile_bdr', tile_bdr)
+ call mpas_pool_get_config(mgr % pool, 'tile_x', tile_nx)
+ call mpas_pool_get_config(mgr % pool, 'tile_y', tile_ny)
+
+ do i = 1, nCells
+     !
+     ! Load the tile at each cell center. This insures all cells receive values
+     !
+     if (nhs(i) == 0) then
+         tile => null()
+         ierr = mgr % get_tile(latCell(i), lonCell(i), tile)
+         if (ierr /= 0 .or. .not. associated(tile)) then
+             call mpas_log_write('Could not get tile that contained cell $i', intArgs=(/i/), messageType=MPAS_LOG_CRIT)
+         end if
+
+         ierr = mgr % push_tile(tile)
+         if (ierr /= 0) then
+             call mpas_log_write("Error pushing this tile onto the stack: "//trim(tile%fname), messageType=MPAS_LOG_CRIT)
+         end if
+     end if
+
+     !
+     ! Process each tile by removing it from the stack. Determine the closest cell center to each tile
+     ! pixel by using a KD search and assign its value to that cell.
+     !
+     do while (.not. mgr % is_stack_empty())
+         tile => mgr % pop_tile()
+
+         if (tile % is_processed) then
+             cycle
+         end if
+
+         call mpas_log_write('Processing tile: '//trim(tile%fname))
+
+         all_pixels_mapped_to_halo_cells = .true.
+
+         do j = 1 + tile_bdr, tile_ny + tile_bdr, 1
+            do k = 1 + tile_bdr, tile_nx + tile_bdr, 1
```

Comment:
As we discussed, let's switch to `i` as the loop variable for the x-direction, and `j` as the loop variable for the y-direction. I think the `, 1` loop increments could also be removed, unless we plan to implement sub-sampling.